Remotion/ObjectBinding/Web/UI/Controls/BocListImplementation/Rendering/BocValidationErrorIndicatorColumnRenderer.cs

```diff
@@ -0,0 +1,100 @@
+  {
+    private const string c_nonBreakingSpace = " ";
+    private const string c_validationIndicatorIcon = "sprite.svg#ValidationError";
+
+    private readonly IInfrastructureResourceUrlFactory _infrastructureResourceUrlFactory;
+    private readonly IBocListValidationSummaryRenderer _validationSummaryRenderer;
+
+    public BocValidationErrorIndicatorColumnRenderer (
+        IResourceUrlFactory resourceUrlFactory,
+        IRenderingFeatures renderingFeatures,
+        BocListCssClassDefinition cssClasses,
+        IFallbackNavigationUrlProvider fallbackNavigationUrlProvider,
+        IInfrastructureResourceUrlFactory infrastructureResourceUrlFactory,
+        IBocListValidationSummaryRenderer validationSummaryRenderer)
+        : base(resourceUrlFactory, renderingFeatures, cssClasses, fallbackNavigationUrlProvider)
+    {
+      ArgumentUtility.CheckNotNull(nameof(infrastructureResourceUrlFactory), infrastructureResourceUrlFactory);
+      ArgumentUtility.CheckNotNull(nameof(validationSummaryRenderer), validationSummaryRenderer);
+
+      _infrastructureResourceUrlFactory = infrastructureResourceUrlFactory;
+      _validationSummaryRenderer = validationSummaryRenderer;
+    }
+
+    protected override string GetAdditionalCssClassForTitleCell (BocCellAttributeRenderingContext<BocValidationErrorIndicatorColumnDefinition> renderingContext, in BocTitleCellRenderArguments arguments)
+    {
+      return CssClasses.TitleCellValidationFailureIndicator;
+    }
+
+    protected override string GetAdditionalCssClassForDataCell (BocCellAttributeRenderingContext<BocValidationErrorIndicatorColumnDefinition> renderingContext, in BocDataCellRenderArguments arguments)
+    {
+      return CssClasses.DataCellValidationFailureIndicator;
+    }
+
+    protected override void RenderCellContents (BocColumnRenderingContext<BocValidationErrorIndicatorColumnDefinition> renderingContext, in BocDataCellRenderArguments arguments)
+    {
+      ArgumentUtility.CheckNotNull(nameof(renderingContext), renderingContext);
+
+      var bocListValidationFailureRepository = renderingContext.Control.ValidationFailureRepository;
+      var validationFailures = bocListValidationFailureRepository.GetUnhandledValidationFailuresForDataRowAndContainingDataCells(arguments.BusinessObject, false);
+      if (validationFailures.Count > 0)
+      {
+        renderingContext.Writer.AddAttribute(HtmlTextWriterAttribute.Class, CssClasses.Content);
+        renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Div);
+
+        var iconInfo = new IconInfo(
+            _infrastructureResourceUrlFactory.CreateThemedResourceUrl(ResourceType.Image, c_validationIndicatorIcon).GetUrl(),
+            new Unit(16, UnitType.Pixel),
+            new Unit(16, UnitType.Pixel));
+        iconInfo.Render(renderingContext.Writer, renderingContext.Control);
```

Comment:
Please check how it's done in BocList.GetValidationErrorMarker

There's actually no reason (except effiiciency) that precludes us from doing `BocList.GetValidationErrorMarker().Render(writer)`
That way we have no duplication. One day, I want to simplify / move the semantics. Until then, it's probably best we try to avoid duplication.

Reply:
Except that it is `protected` ^^